app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,23 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(60) do
+        self.ldap_con.valid_user?("")
```

Comment:
@domcleal  I didn't notice, my bad. I'm on it. 
The test function in ldap_fluff still works, so I think this PR can merged before the improvement.